### DISCUSSION TITLE
A question about the class ElementIntegralVariablePostprocessor

### DISCUSSION THREAD

**Discussion by TJT-post95** [idaholab/moose#22472] [Q&A General] [answered]
Dear MOOSE:
In the 'Example 20 - UserObjects', the `BlockAverageValue ` which derived from class `ElementIntegralVariablePostprocessor `has been applied. It could be found that one of the parent classes of `ElementIntegralVariablePostprocessor `is `MooseVariableInterface<Real>`. However, the member attributes `_u` and `_grad_u` are initialized by `coupledValue()` and `coupledGradient()` respectively. And `addMooseVariableDependency(&mooseVariableField())` could be replaced by `addMooseVariableDependency(getCoupledMooseVars())` to make the same effect. So it seems that do not inherit class `MooseVariableInterface<Real>` the `ElementIntegralVariablePostprocessor `also could work. Does `ElementIntegralVariablePostprocessor `inherit class `MooseVariableInterface<Real>` have other considerations, is it necessary?
Thank you!

**Comment by GiudGiud** [comment, marked as answer]
Hello

you have to consider the needs of all the subclasses and potential subclasses of this class. 

As you can see in the base classes like UserObject or ElementUserObject, we inherit a lot of interfaces to be ready for whatever a derived class needs. 

I can check tomorrow this particular situation 

Guillaume

**Reply by GiudGiud** [reply]
I agree we dont need it in this class.

**Reply by TJT-post95** [reply]
Got. Thank you～
